dbt_subprojects/hourly_spellbook/models/_project/cow_protocol/polygon/cow_protocol_polygon_batches.sql

```diff
@@ -0,0 +1,119 @@
+{{  config(
+        schema = 'cow_protocol_polygon',
+        alias = 'batches',
+        materialized='incremental',
+        partition_by = ['block_date'],
+        unique_key = ['tx_hash'],
+        on_schema_change='sync_all_columns',
+        file_format ='delta',
+        incremental_strategy='merge',
+        incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')],
+        post_hook='{{ expose_spells(blockchains = \'["polygon"]\',
+                                    spell_type = "project",
+                                    spell_name = "cow_protocol",
+                                    contributors = \'["harisang"]\') }}'
+    )
+}}
+
+with
+solvers_ranked as (
+    select
+        t.block_date,
+        t.block_number,
+        t.tx_hash,
+        t."from" as solver,
+        solvers.name,
+        row_number() over (partition by t.tx_hash order by t.trace_address asc nulls first) as rn
+    from {{ source('polygon', 'traces') }} t
+        inner join {{ ref('cow_protocol_polygon_solvers') }} solvers
+            on t."from" = solvers.address
+    {% if is_incremental() %}
+    where {{ incremental_predicate('block_time') }}
+    {% endif %}
```

Comment:
```suggestion
    {% else %}
    where block_time >= timestamp '2023-08-03 16:25' --first block_time observed
    {% endif %}
```

Reply:
adding this for historical run to see if it helps initial build runtime. found via first date in CI table built:
```
select min(block_time)
from test_schema.git_dunesql_6b5dceb_cow_protocol_polygon_batches
limit 10
```